Decred Wallet/Features/Wallet Utils/SyncManager.swift

```diff
@@ -0,0 +1,152 @@
+                noConnectionAlert.addAction(UIAlertAction(title: LocalizedStrings.ok, style: .default, handler: nil))
+                AppDelegate.shared.window?.rootViewController?.present(noConnectionAlert, animated: false, completion: self.checkSyncPermission)
+                self.connectedToNetwork?(false)
+            }
+        } else {
+            self.checkSyncPermission()
+        }
+    }
+    
+    func checkSyncPermission() {
+        if AppDelegate.shared.reachability.connection == .none {
+            self.syncNotStartedDueToNetwork()
+        } else if AppDelegate.shared.reachability.connection == .wifi || Settings.syncOnCellular {
+            self.startSync()
+        } else {
+            self.requestPermissionToSync()
+        }
+    }
+    
+    func requestPermissionToSync() {
+        
+        let syncConfirmationDialog = UIAlertController(title: LocalizedStrings.internetConnectionRequired, message: LocalizedStrings.cannotSyncWithoutNetworkConnection, preferredStyle: .alert)
+        
+        syncConfirmationDialog.addAction(UIAlertAction(title: LocalizedStrings.allowOnce, style: .default, handler: { action in
+            self.startSync()
+        })) // Allow once selected
+        
+        syncConfirmationDialog.addAction(UIAlertAction(title: LocalizedStrings.alwaysAllow, style: .default, handler: { action in
+            Settings.setValue(true, for: Settings.Keys.SyncOnCellular)
+            self.startSync()
+        })) // Always allow even on cellular selected
+        
+        syncConfirmationDialog.addAction(UIAlertAction(title: LocalizedStrings.notNow, style: .cancel, handler: { action in
+            self.syncNotStartedDueToNetwork()
+        })) // Not now selected. Do not sync
+        
+        DispatchQueue.main.async {
+            AppDelegate.shared.window?.rootViewController?.present(syncConfirmationDialog, animated: true, completion: nil)
+        }
+    }
+    
+    func syncNotStartedDueToNetwork() {
+        AppDelegate.walletLoader.syncer.deRegisterSyncProgressListener(for: "\(self)")
+        AppDelegate.walletLoader.wallet?.cancelSync()
+        
+        // Allow 0.5 seconds for sync cancellation to complete before setting up wallet.
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+            AppDelegate.walletLoader.syncer.assumeSyncCompleted()
+            self.onSyncEndedWithError(LocalizedStrings.connectToWiFiToSync)
+            self.connectedToNetwork?(false)
+        }
+    }
+    
+    func startSync(isRestarting: Bool = false) {
+        AppDelegate.walletLoader.syncer.registerSyncProgressListener(for: "\(self)", self)
+        DispatchQueue.main.asyncAfter(deadline: .now()) {
+            self.connectedToNetwork?(true)
+        }
+        _ = isRestarting ? AppDelegate.walletLoader.syncer.restartSync() : AppDelegate.walletLoader.syncer.beginSync()
```

Comment:
Use an if statement instead. Neither functions return a value, this could be misleading.

Reply:
resolved in aa27838

Reply:
added a comment to provide clarity and resolved in aa27838